src/catalogue/category/catalogueCategoryDialog.component.tsx

```diff
@@ -12,636 +16,861 @@ import {
+        </FormHelperText>
+      )}
+    </>
+  );
+};
 
-      if (parentId !== null) {
-        catalogueCategory = {
-          ...catalogueCategory,
-          parent_id: parentId,
+const CatalogueCategoryDialog = (props: CatalogueCategoryDialogProps) => {
+  const {
+    open,
+    onClose,
+    parentId,
+    requestType,
+    duplicate,
+    selectedCatalogueCategory,
+    resetSelectedCatalogueCategory,
+  } = props;
+
+  const initialCatalogueCategory: AddCatalogueCategoryWithPlacementIds =
+    React.useMemo(() => {
+      if (
+        (!selectedCatalogueCategory &&
+          (requestType === 'post' || !duplicate)) ||
+        !selectedCatalogueCategory
+      )
+        return {
+          name: '',
+          parent_id: null,
+          is_leaf: 'false', // Use 'false' as a string instead of a boolean
+          properties: undefined,
         };
-      }
-      if (updatedProperties) {
-        catalogueCategory = {
-          ...catalogueCategory,
-          properties: updatedProperties,
-        };
-      }
 
-      addCatalogueCategory(trimStringValues(catalogueCategory))
+      return transformToAddCatalogueCategoryWithPlacementIds(
+        selectedCatalogueCategory
+      );
+    }, [requestType, duplicate, selectedCatalogueCategory]);
+
+  const formMethods = useForm<AddCatalogueCategoryWithPlacementIds>({
+    resolver: zodResolver(CatalogueCategorySchema),
+    defaultValues: initialCatalogueCategory,
+  });
+
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+    watch,
+    control,
+    setError,
+    clearErrors,
+    reset,
+    resetField,
+    setValue,
+  } = formMethods;
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: 'properties',
+  });
+
+  const isLeaf = watch(`is_leaf`);
+
+  // Load the values for editing.
+  React.useEffect(() => {
+    reset(initialCatalogueCategory);
+  }, [initialCatalogueCategory, reset]);
+
+  const { mutateAsync: postCatalogueCategory, isPending: isAddPending } =
+    usePostCatalogueCategory();
+  const { mutateAsync: patchCatalogueCategory, isPending: isEditPending } =
+    usePatchCatalogueCategory();
+
+  const handleClose = React.useCallback(() => {
+    reset();
+    clearErrors();
+    onClose();
+    resetSelectedCatalogueCategory();
+  }, [clearErrors, onClose, reset, resetSelectedCatalogueCategory]);
+
+  const handleAddCatalogueCategory = React.useCallback(
+    (data: CatalogueCategoryPost) => {
+      postCatalogueCategory(data)
         .then(() => handleClose())
-        .catch((error) => {
+        .catch((error: AxiosError) => {
           const response = error.response?.data as APIError;
           if (response && error.response?.status === 409) {
-            setNameError(response.detail);
+            setError('name', {
+              message:
+                'A Catalogue category with the same name already exists within the same parent Catalogue category. Please enter a different name.',
```

Comment:
I think this should either be `catalogue category` or `Catalogue Category` depending on how we used it elsewhere. The backend we moved to all lowercase.